### PR TITLE
API changes for dynamic engagement pages

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-500

*Description of changes:*
    - Implemented endpoints for dynamic engagement pages, including summary and custom sections.
    - Default behavior ensures that each engagement has a dynamic summary page.
    - Introduced logic to migrate existing content and rich content for engagements to the summary table.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/services/engagement_content_service.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+        sort_index = EngagementContentService._find_higest_sort_index(engagement_id)
+
+        engagement_content_data['sort_index'] = sort_index + 1
+        created_content = EngagementContentService._create_content(engagement_id, engagement_content_data)
+        created_content.commit()
+        return EngagementContentSchema().dump(created_content)
+
+    @staticmethod
+    def _find_higest_sort_index(engagement_id):
```
Comment (Baelx) [inline met-api/src/met_api/services/engagement_content_service.py:44]: In other words, does this method find the "last" engagement content with regards to sort order?

File: met-api/src/met_api/services/engagement_custom_content_service.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from met_api.services import authorization
+from met_api.utils.roles import Role
+
+
+class EngagementCustomContentService:
+    """Engagement custom content management service."""
+
+    @staticmethod
+    def get_custom_content(content_id):
+        """Get map by engagement custom content id."""
```
Comment (Baelx) [inline met-api/src/met_api/services/engagement_custom_content_service.py:13]: What do you mean by "get map" here?

**Review by Baelx — APPROVED** [review]
I had some minor questions/remarks here and there but this is approved!

File: met-api/migrations/versions/e2625b0d07ab_add_tables_for_dynamic_eng_pages.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        SELECT id, content, rich_content
+        FROM engagement
+    """)
+    engagements = result.fetchall()
+
+    # Initialize a counter variable
+    counter = 1
+
+    # Iterate through each engagement record and insert into new tables
+    for engagement in engagements:
```
[the remark this answers is not in the document]
Comment (VineetBala-AOT) [reply met-api/migrations/versions/e2625b0d07ab_add_tables_for_dynamic_eng_pages.py]: Thanks, I have incorporated the suggested changes into the logic.

File: met-api/src/met_api/services/engagement_content_service.py (hunk 1 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/services/engagement_content_service.py:44]: Yes

File: met-api/src/met_api/services/engagement_custom_content_service.py (hunk 2 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/services/engagement_custom_content_service.py]: Thanks for noticing this, I updated the comments here.

**Pull request merged**